tests/test_merge.py

```diff
@@ -51,33 +52,226 @@ def test_merge_branch(wst, capsys):
     assert out.split('\n')[0] == 'Merging 3.0.x into master'
 
 
-def test_merge_branch_with_whitelist(wst, capsys):
+def make_commit(name, skip=False):
+    run('touch {}.xml'.format(name))
+    run('git add -A')
+    msg = name
+
+    if skip:
+        msg = "[skip]{}".format(msg)
+
+    run('git commit -m {}'.format(msg))
+    sha = run('git rev-parse --verify HEAD', return_output=True).strip()
+    return sha
+
+
+# For all test_merge_branch tests below, we cannot assert commits in the log line-by-line because the ordering of the
+# commit messages is not guaranteed because the timestamps on the commits are all the same. Adding sleep statements
+# allows the order to be guaranteed and replicable but will slow down the tests.
+
+
+def test_merge_branch_skip_last(wst, capsys):
```

Comment:
lets put method descriptions for all our test methods. brief description of what we are testing for.